pennylane/gradients/gradient_transform.py

```diff
@@ -117,6 +119,14 @@ def default_qnode_wrapper(self, qnode, targs, tkwargs):
         cjac_fn = qml.transforms.classical_jacobian(qnode, expand_fn=expand_invalid_trainable)
 
         def jacobian_wrapper(*args, **kwargs):
+            if not qml.math.get_trainable_indices(args):
+                warnings.warn(
+                    "Attempted to compute the gradient of a QNode with no trainable parameters. "
+                    "If this is unintended, please add trainable parameters in accordance with "
+                    "your autodiff framework."

```

Comment:
Minor: "your" could imply to a user the framework that they have installed/configured, etc.

If the suggestion is accepted, this would be recurring and applicable at other spots too.
```suggestion
                    "the chosen auto differentiation framework."
```